src/main/java/org/sagebionetworks/database/semaphore/CountingSemaphoreImpl.java

```diff
@@ -49,6 +52,15 @@
 			.getLogger(CountingSemaphoreImpl.class);
 
 	private static final String SQL_CLEAR_ALL_LOCKS = "UPDATE "+ TABLE_SEMAPHORE_LOCK+" SET TOKEN = NULL WHERE LOCK_KEY IS NOT NULL";
+
+	private static final String SQL_EXISTS_UNEXPIRED_LOCK =
+			"SELECT EXISTS (" +
+			"SELECT *" +
+			" FROM " + TABLE_SEMAPHORE_LOCK +
+			" WHERE " + COL_TABLE_SEM_LOCK_LOCK_KEY + " = ?" +
+			" AND " + COL_TABLE_SEM_LOCK_TOKEN + " IS NOT NULL " +
+			" AND " + COL_TABLE_SEM_LOCK_EXPIRES_ON + " >= CURRENT_TIMESTAMP" +
+			")";

```

Comment:
can we add a limit 1 to the inner query?  There is no need for the database return more than one row.